src/App.jsx

```diff
@@ -14,33 +16,89 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_ALPHABETICALLY = 'Sort alphabetically';
+const SORT_BY_LENGTH = 'Sort by length';
+const REVERSE = false;
+const RESET = '';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function arraysEqual(a, b) {
+  return a.every((val, index) => val === b[index]);
+}
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(REVERSE);
+  const visibleGoods = [...goodsFromServer];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  visibleGoods.sort((good1, good2) => {
+    switch (sortField) {
+      case SORT_ALPHABETICALLY:
+        return good1.localeCompare(good2);
+      case SORT_BY_LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (reversed) {
+    visibleGoods.reverse();
+  }
```

Comment:
Create a function to do it and move it outside the component

Reply:
it was tough :D